clang-tools-extra/clang-doc/assets/index.js

```diff
@@ -80,8 +80,8 @@ function createIndex(Index) {
 
 // Runs after DOM loads
 document.addEventListener("DOMContentLoaded", function() {
-  // JsonIndex is a variable from another file that contains the index
-  // in JSON format
-  var Index = JSON.parse(JsonIndex);
-  createIndex(Index);
+  // LoadIndex is an asynchronous function that will be generated clang-doc
+  // it ensures that the function call will not blocked as soon the page loads
+  // since the index object are often huge and can contain thousands of lines
+  LoadIndex().then((Index) => { createIndex(Index); });
```

Comment:
```suggestion
  // LoadIndex is an asynchronous function that will be generated clang-doc.
  // It ensures that the function call will not block as soon the page loads,
  // since the index object are often huge and can contain thousands of lines.
  LoadIndex().then((Index) => { createIndex(Index); });
```

Reply:
@PeterChou1, once this comment is fixed, I'm happy to land this.